kube-runtime/src/utils/event_flatten.rs

```diff
@@ -36,22 +36,26 @@ where
             if let Some(item) = me.queue.next() {
                 break Some(Ok(item));
             }
-            break match ready!(me.stream.as_mut().poll_next(cx)) {
-                Some(Ok(Event::Applied(obj))) => Some(Ok(obj)),
+            let var_name = match ready!(me.stream.as_mut().poll_next(cx)) {
+                Some(Ok(
+                    Event::Applied(obj) | Event::RestartedApplied(obj) | Event::RestartedDeleted(obj),
```

Comment:
the `RestartDeleted` part here needs to be handled inside the match below respecting the `me.emit_deleted`

Reply:
done in https://github.com/kube-rs/kube/pull/1494/commits/eda47f2fe98a48df5206820996c43c767252fef8